open_src/influx/meta/data.go

```diff
@@ -1811,6 +1811,28 @@ func (data *Data) CreateSubscription(database, rp, name, mode string, destinatio
 
 // DropSubscription removes a subscription.
 func (data *Data) DropSubscription(database, rp, name string) error {
+	// Drop all subscriptions
+	if database == "" {
+		for _, db := range data.Databases {
+			for _, rp := range db.RetentionPolicies {
+				rp.Subscriptions = rp.Subscriptions[:0]
+			}
+		}
+		return nil
+	}
+
+	// Drop all subscriptions on the Specified db
+	if name == "" {
+		db, ok := data.Databases[database]
+		if !ok {
+			return ErrDatabaseNotExists
+		}
+		for _, rp := range db.RetentionPolicies {
+			rp.Subscriptions = rp.Subscriptions[:0]
+		}
+		return nil
+	}
+
```

Comment:
how to deal with:
`DROP SUBSCRIPTION "<subscription_name>" ON "<db_name>"`

Reply:
In this case, the retention policy will be replaced by "autogen" during the parsing phase. "autogen" is the default retention policy. See https://github.com/openGemini/openGemini/blob/subscribution_develop/open_src/influx/influxql/sql.y#L2647